packages/transformer/src/IModelTransformer.ts

```diff
@@ -2939,118 +2948,103 @@ export class IModelTransformer extends IModelExportHandler {
-        const aspects: ExternalSourceAspect[] =
-          this.sourceDb.elements.getAspects(
-            instId,
-            ExternalSourceAspect.classFullName
-          ) as ExternalSourceAspect[];
-        for (const aspect of aspects) {
-          // look for aspect where the ecInstanceId = the aspect.element.id
-          if (
-            aspect.element.id === instId &&
-            aspect.scope.id === this.targetScopeElementId
-          )
-            identifierValue = aspect.identifier;
-        }
-        // Think I need to query the esas given the instId.. not sure what db to do it on though.. soruce or target.. or provenance?
-        // I need to know the id of the element dpeneding on which db its stored in.
-      }
-      if (queryCanAccessProvenance && !identifierValue) {
-        if (mapOfDeletedElemIdToScopeEsas.get(instId) !== undefined)
-          identifierValue =
-            mapOfDeletedElemIdToScopeEsas.get(instId)!.Identifier;
-      }
-      const targetId =
-        (queryCanAccessProvenance && identifierValue) ||
-        // maybe batching these queries would perform better but we should
-        // try to attach the second db and query both together anyway
-        (sourceElemFedGuid &&
-          this._queryElemIdByFedGuid(this.targetDb, sourceElemFedGuid)) ||
-        // FIXME<MIKE>: describe why it's safe to assume nothing has been deleted in provenanceDb
-        this._queryProvenanceForElement(instId);
 
-      // since we are processing one changeset at a time, we can see local source deletes
-      // of entities that were never synced and can be safely ignored
-      const deletionNotInTarget = !targetId;
-      if (deletionNotInTarget) return;
-      this.context.remapElement(instId, targetId);
-      // If an entity insert and an entity delete both point to the same entity in target iModel, that means that entity was recreated.
-      // In such case an entity update will be triggered and we no longer need to delete the entity.
-      if (alreadyImportedElementInserts.has(targetId)) {
-        this.exporter.sourceDbChanges?.element.deleteIds.delete(instId);
-      }
-      if (alreadyImportedModelInserts.has(targetId)) {
-        this.exporter.sourceDbChanges?.model.deleteIds.delete(instId);
-      }
-    } else {
-      // is deleted relationship
-      const classFullName = change.$meta?.classFullName;
-      const sourceIdOfRelationshipInSource = change.SourceECInstanceId;
-      const targetIdOfRelationshipInSource = change.TargetECInstanceId;
-      const [sourceIdInTarget, targetIdInTarget] = [
-        sourceIdOfRelationshipInSource,
-        targetIdOfRelationshipInSource,
-      ].map((id) => {
+    const changedInstanceId = change.ECInstanceId;
+    const sourceIdOfRelationshipInSource = change.SourceECInstanceId;
+    const targetIdOfRelationshipInSource = change.TargetECInstanceId;
+    const classFullName = change.$meta?.classFullName;
+    const idsInSourceDb = isRelationship
```

Comment:
it feels like if you merge all the `isRelationship` branches into one branch it would be simpler. You would probably want to have a local function for getting the target id of an element from any source element and use it there.

Reply:
added local function 'getTargetIdFromSourceId'